src/js/tabs/osd.js

```diff
@@ -1811,8 +1811,10 @@ OSD.msp = {
 
         d.state = {};
         d.state.haveSomeOsd = (d.flags != 0)
-        d.state.haveMax7456Video = bit_check(d.flags, 4) || (d.flags == 1 && semver.lt(CONFIG.apiVersion, "1.34.0"));
-        d.state.isMax7456Detected = bit_check(d.flags, 5) || (d.state.haveMax7456Video && semver.lt(CONFIG.apiVersion, "1.43.0"));
+        d.state.haveMax7456Configured = bit_check(d.flags, 4) || (d.flags == 1 && semver.lt(CONFIG.apiVersion, "1.34.0"));
+        d.state.haveFrSkyOSDConfigured = semver.gt(CONFIG.apiVersion, "1.43.0") && bit_check(d.flags, 3);
+        d.state.haveMax7456FontDeviceConfigured = d.state.haveMax7456Configured || d.state.haveFrSkyOSDConfigured;
+        d.state.isMax7456FontDeviceDetected = bit_check(d.flags, 5) || (d.state.haveMax7456FontDeviceConfigured && semver.lt(CONFIG.apiVersion, "1.43.0"));
```

Comment:
This seems to be wrong - Betaflight configurator will only manage fonts if a MAX7456 is used for OSD - there is no point to display the font manager and give the user the wrong and misleading impression of this being different in any other case.

Reply:
FrSkyOSD uses the same font format as MAX7456, that's why the "Upload Font" button should show up. Users can change the font as if it was a MAX7456.

Reply:
I think this is the wrong approach. If FrSky wants to sell an OSD device they should pay for development and maintenance of the tools to configure the device they sell.
Some problems I can see with this, especially considering that FrSkyOSD is a closed source commercial project over which Betaflight has got no control:
- when there are problems / issue tickets opened by users trying to upload fonts to a FrSkyOSD device with Betaflight configurator and failing, will FrSky pay for the time spent by volunteers of the Betaflight open source project to help these users? I do not think it is appropriate for a commercial entity to expect volunteers in an open source project to sacrifice their time for an entity making money from them;
- when Betaflight decides to move to a different format for font files (like what is proposed in #1174), this will break the ability for all customers of the FrSkyOSD to use Betaflight configurator to upload font to the hardware they paid for. Will FrSky pay the Betaflight project to not break their customers' ability to configure the device they bought from FrSky?
So all in all I think it is best if FrSky developed and published their own application to configure FrSkyOSD devices.

Reply:
I think this is all coming from a misunderstanding. Let me try to clear that up.

FrSkyOSD, by design, takes the same fonts as MAX7456 and this won't ever change. You seem to think that FrSkyOSD uses an approach like DJI's and does all the drawing by itself, but that's not the case. FrSky's exposes a canvas and all the drawing is driven by the flight controller. The OSD does not care about the contents of each character and each flight controller firmware will have its own font files, just as it happens with MAX7456. As long as Betaflight keeps supporting MAX7456, there's no additional overhead in supporting font uploads for the FrSky OSD. In fact, font uploads did work fine when support for FrSkyOSD was merged. It was broken later in https://github.com/betaflight/betaflight/pull/9158 and https://github.com/betaflight/betaflight-configurator/pull/1818, because the new code explicitly checks for the OSD type and wether the hardware has been actually detected instead of checking for a device that takes MAX7456 fonts. All this PR does is teach the configurator that FrSkyOSD does take the same font files as MAX7456. Nothing else.

Regarding your question about #1174, the short answer is that it will work fine with FrSkyOSD. The long answer is that the OSD uses the first 54 bytes as visible font data, but you can upload and read back the whole 64 bytes per character. Again, exactly the same as the MAX7456.

Speaking about firmware updates, these are a totally different thing that font uploads. Of course, firmware updates for FrSkyOSD will be handled by a separate FrSky tool, but font files are entirely the responsibility of the flight controller software driving the OSD just as with, you guessed it, MAX7456. It wouldn't make any sense to need an external tool to load the font files, just like it wouldn't make any sense to ask users to use a MAXIM tool to upload their fonts for the MAX7456 because they do depend on the software driving the FrSkyOSD, not on the FrSkyOSD itself.

Finally, I don't understand the fuss about the hardware being closed. All hardware on board is basically closed. In fact, you'd be hard pressed to find a single device that is open in most user setups. CPUs, IMUs, MAGs, MAX756 are all closed just to name a few. Heck, even the VTX hardware and firmware is closed source and those have quite a lot more code to support and interact with them. No one is asking Betaflight to give this hardware any special treatment, just the same as every other device.

Reply:
No, there is definitely no misunderstanding. FrSky is trying to freeload on the work of volunteers to get out of providing a configuration app for the hardware that they are selling. If they want to outsource this to Betaflight we can talk about this, and I am sure we will be able to agree on what cost they will have to cover for our efforts. If not then they should provide their own app to configure the hardware they are selling, like everybody else.

Reply:
You're grasping at straws. FrSky is not "outsourcing" configuration of this OSD to BF anymore than MAXIM is "outsourcing" configuration of the MAX7456. Have you looked at the code in this PR? It's literally 10 lines of code just indicating FrSkyOSD as compatible with MAX7456 font files.

What are those manufacturers you mention that are providing external tools to configure the interactions between their hardware and Betaflight? I don't see e.g. any TBS tools for configuring their SmartAudio VTX, but Betaflight does have not just full support for configuring them, but also [a page in the Wiki](https://github.com/betaflight/betaflight/wiki/Unify-Smartaudio) with instructions for users.

Reply:
It seems to me as if the FrSky OSD is a commercial product implementing a (commercial, but not closed) communication standard, that is defined as the MAX7456 interface. It is in the interest of BetaFlight users that this standard, and all devices supporting it, are available for use within BetaFlight.

If a tool was required specifically to configure or interact with the FrSkyOSD, outside of communications that BetaFlight requires to function in general, then this could be considered out of scope for the BetaFlight project and potentially questionable to implement for the political reasons discussed above. However from my technical understanding it seems that the configuration of the display module (be it MAX7456 or MAX7456-alike), does fall within the scope of the code driving it (in this case BetaFlight), since the chosen byte-to-char semantic mapping needs to be consistent between the configured font and the intended usage by the application.  
It could be argued that font managers don't have anything to do with the BetaFlight configurator in general, and this may be true, but having it integrated in the configurator is clearly a design convenience that appealed to whoever put it there in the first place. I don't see how this would change due to another entity adding a product compatible with existing tooling.

Reply:
@fiam:

> FrSky is not "outsourcing" configuration of this OSD to BF anymore than MAXIM is "outsourcing" configuration of the MAX7456.

Sorry, I could not find any information of a MAXIM selling hardware for RC pilots - to my knowledge they are only selling integrated circuits. Can you provide links?

> I don't see e.g. any TBS tools for configuring their SmartAudio VTX

Again, I could not find any information on persistent configuration that can be made for SmartAudio devices besides in-flight changes like changing frequency and output power. Can you provide links for documentation of the configuration you are referring to?

For devices from other manufacturers that can be configured, like IRC tramp, they are providing their own tooling to allow their customers to change the settings on their hardware, like enabling race mode or selecting the compliance zone.